### ISSUE TITLE
Support for custom Automatic Type Acquisition scopes.

### ISSUE THREAD

**Issue by AaronFriel** [Microsoft/TypeScript#23952]
## Search Terms

Automatic type acquisition scopes, custom type acquisition, acquire types from custom scope

## Suggestion

The ability to specify in `.tsconfig` and/or IDE settings additional scopes to search beyond `@types` for type acquisition.

## Use Cases

Private repositories with large amounts of JavaScript code and a small number of TypeScript developers who wish to produce accurate typings, and a high barrier to entry to modifying/adding declaration files to the JavaScript projects. This mirrors the public ecosystem with say, `react` and `@types/react`.

## Examples

A new string array parameter of scopes to search for:

```json
{
  "typeAcquisition": {
    "searchScopes": [
      "@companyname-types",
      "@types"
    ]
  }
}
```

This would also allow overriding the `@types` scope. The first scope with a matching declaration should win.

## Checklist

My suggestion meets these guidelines:
 [X] This wouldn't be a breaking change in existing TypeScript / JavaScript code
 [X] This wouldn't change the runtime behavior of existing JavaScript code
 [X] This could be implemented without emitting different JS based on the types of the expressions
 [X] This isn't a runtime feature (e.g. new expression-level syntax)

**Comment by AaronFriel** [comment]
Commentary: @ahejlsberg told me to make this request, I'd be happy to look into implementing this and turning it into a PR myself, but I am not sure where to begin looking.

**Comment by mhegazy** [comment]
this is not the same as `typeRoot`. typeroots assume the packages has already been downloaded and available on disk. If i understand this proposal correctly, you want the IDE to lookup packages in other registries/scopes as well..

**Comment by AaronFriel** [comment]
@mhegazy, yes, I think "other registries" is a bit out of scope for this request, but yes.
